pyproject.toml

```diff
@@ -43,7 +43,7 @@ PyYAML = {version = "*", optional = true}  # groups = ["user", "dev"]
 requests = {version = "*", optional = true}  # groups = ["user", "dev"]
 
 numpy = {version = ">= 1.12", optional = true}  # groups = ["user", "dev"]
-matplotlib = {version = "*", optional = true}  # groups = ["user", "dev"]
+matplotlib = "3.2"  # groups = ["user", "dev"]
```

Comment:
Would you be able to provide a more detailed reasoning for this?

Reply:
I'll start looking at the matplotlib compatibility issues

> matplotlib 3.3 doesn't play well with sip.setapi but version 3.2 does

https://github.com/ipython/ipython/issues/276 has a bit to say about it, or at least something very strongly related to it.
So there might be some magic in calling sip.setapi with the right version at the right time or doing the import in a particular order